crates/bevy_reflect/src/tuple_struct.rs

```diff
@@ -43,6 +48,60 @@ pub trait TupleStruct: Reflect {
     fn clone_dynamic(&self) -> DynamicTupleStruct;
 }
 
+/// A container for compile-time tuple struct info
+#[derive(Clone, Debug)]
+pub struct TupleStructInfo {
+    type_name: Cow<'static, str>,
+    fields: Vec<UnnamedField>,
```

Comment:
This is immutable upon construction. Capacity and the ability to extend/add fields after construction is unnecessary. Please switch to using Box<[UnnamedField]> instead. Alternatively, if it's possible to make inputs during construction `const`, a `&'static [UnnamedField]`.